### ISSUE TITLE
Adding 1-1 interview using webRTC

### ISSUE THREAD

**Issue by RohankumarM** [codeforcauseorg/edu-client#77]
It takes a lot of money to invest just to prepare yourself for an interview so why not users get a chance to interview each other by sharing their video, audio, and screen in a 1 - 1 fashion. 

We could use technology like WebRTC to implement this feature. The interview process will not be limited to just solving data structures and algorithms related problems but could include much more.

**Comment by Itsaadarsh** [comment]
Yes, @RohankumarM this sounds great as I have previously worked with webRTC technology where I used SFU architecture to build multi-party audio rooms. So if the maintainers want this feature to be integrated into the platform I'd glad to work on it.

And as it is going to have 1-1 room there will not be a load on our servers too as we can make use of P2P architecture (Peer-To-Peer) which is the core fundamental of webRTC.

**Comment by Jeevan-Kiran-Lenka** [comment]
@Itsaadarsh I very much agree with this idea

**Comment by RohankumarM** [comment]
Definitely @Itsaadarsh. I have also have worked with WebRTC and I just love its architecture and I think this feature would be in demand. But before moving ahead, we just need to make sure the UI of this feature is user-friendly because that will be the main factor.

**Comment by Itsaadarsh** [comment]
Hey @Abhishek-kumar09, how do we get started with implementing this feature? Like do we wait for the whole UI/UX to be completed or we just start building this feature on the go? And on the other hand, we would also need to make a few commits on the `edu-server` repository in order to build this feature. Because the P2P architecture requires an initial handshake to take place which will be processed by our server.

**Comment by Abhishek-kumar09** [comment]
You can start working without design, I have added the making of design of interview session page in the task pipeline. It will be available soon.

But for now we can start working on the actual problem without thinking of designs.
>  we would also need to make a few commits on the edu-server repository

yes, sure. It is built on nestJs and typescript. Let me know if you need some help with that.

**Comment by Itsaadarsh** [comment]
Alright then I'll get started right away, and can you also assign this to me?

**Comment by RohankumarM** [comment]
@Itsaadarsh If you need any help you can definitely reach out to me.

**Comment by Abhishek-kumar09** [comment]
Hi @RohankumarM Any specific reason for closing this issue? We should close this once the feature is merged in main code.

**Comment by RohankumarM** [comment]
@Abhishek-kumar09 I hit close by mistake. No worries. Eargerly waiting for the feature to be added.

**Comment by Itsaadarsh** [comment]
@RohankumarM The work has already begun client-side logic is mostly done, I just need to figure out `nestjs` on the server-side and how to use WebSockets in `nestjs` once I figure that out I'll send a PR.

**Comment by RohankumarM** [comment]
@Itsaadarsh Impressive. Did you try using express with nextjs? Would be easier that way I guess.

**Comment by RohankumarM** [comment]
@Itsaadarsh I hope you and your family are in great health. Just reaching to check with you about any updates on this feature? Just curious. :)
